app/static/js/admin/media/upload.js

```diff
@@ -23,8 +23,47 @@ $(function() {
 
     /* let the `btn-upload` div act as the submit button */
     $(".btn-upload").click(function () {
-        $('form.upload-form').submit();
-        return false;
+        var formData = new FormData($('.upload-form')[0]);
+        $.ajax({
+            url: '/admin/media/upload',  //server script to process data
+            type: 'POST',
+            xhr: function() {  // custom xhr
+                myXhr = $.ajaxSettings.xhr();
+                return myXhr;
+            },
+            //Ajax events
+            success: function(data, textStatus, jqXHR) {
+                var response = jQuery.parseJSON(jqXHR.responseText);
+                if (response.extension == null && response.filename == null){
+                    $('#images-ajax-loadpoint').load("/admin/media/view", function(response, status){
+                        if (status == "error"){
+                            $('error-message').text("Sorry, there was an error loading the images.");
+                        }
+                    });
+                    $('.error-message').hide();
+                }
+                else {
+                    if (response.extension == null)
```

Comment:
Why do we show populate the error message with the text and/or extension?  Is there a more meaningful error to show if we get here?